src/components/DeploymentList.vue

```diff
@@ -0,0 +1,352 @@
+<template>
+  <p-content class="deployment-list">
+    <p-list-header sticky>
+      <ResultsCount v-if="selectedDeployments.length == 0" label="Deployment" :count="total" />
+      <SelectedCount v-else :count="selectedDeployments.length" />
+      <DeploymentsDeleteButton v-if="can.delete.deployment" :selected="selectedDeployments.map(deployment => deployment.id)" small @delete="deleteDeployments" />
+
+      <template #controls>
+        <SearchInput v-model="nameLike" small placeholder="Search deployments..." class="deployment-list__search-input" label="Search deployments" />
+        <DeploymentTagsInput v-model:selected="filter.deployments.tags.name" small multiple />
+      </template>
+
+      <template #sort>
+        <p-select v-model="filter.sort" small :options="deploymentSortOptions" />
+      </template>
+    </p-list-header>
+
+    <p-table
+      :selected="can.delete.deployment ? selectedDeployments : undefined"
+      :data="deployments"
+      :columns="columns"
+      :header-classes="columnClasses"
+      :column-classes="columnClasses"
+      :row-classes="rowClasses"
+      class="deployment-list__table"
+      @update:selected="selectedDeployments = $event"
+    >
+      <template #status="{ row }">
+        <DeploymentStatusBadge :deployment="row" small />
+      </template>
+
+      <template #schedules-heading="{ column }">
+        <div class="deployment-list__schedules-heading">
+          {{ column.label }}
+        </div>
+      </template>
+
+      <template #tags-heading="{ column }">
+        <div class="deployment-list__tags-heading">
+          {{ column.label }}
+        </div>
+      </template>
+
+      <template #action-heading>
+        <span />
+      </template>
+
+      <template #deployment="{ row }">
+        <div class="deployment-list__deployment">
+          <router-link :to="routes.deployment(row.id)" class="deployment-list__name">
+            {{ row.name }}
+          </router-link>
+
+          <FlowPopover :flow-id="row.flowId" class="deployment-list__flow-name" />
```

Comment:
nit: some of these elements have classes but no styles. Probably don't need them.